storage/local/storage_test.go

```diff
@@ -1405,3 +1410,50 @@ func verifyStorage(t testing.TB, s *memorySeriesStorage, samples clientmodel.Sam
 	}
 	return result
 }
+
+func TestAppendOutOfOrder(t *testing.T) {
+	s, closer := NewTestStorage(t, 1)
+	defer closer.Close()
+
+	m := clientmodel.Metric{
+		clientmodel.MetricNameLabel: "out_of_order",
+	}
+
+	for _, t := range []int{0, 2, 2, 1} {
```

Comment:
Can we set the Value to `i` so down below it is clearer which sample was actually inserted?